STACpopulator/populator_base.py

```diff
@@ -87,15 +71,26 @@ def item_properties_model(self):
         models.STACItemProperties."""
         pass
 
+    @property
+    @abstractmethod
+    def item_geometry_model(self):
+        """In derived classes, this property should be defined as a pydantic data model that derives from
+        models.STACItemProperties."""
+        pass
+
+    @abstractmethod
+    def create_stac_item(self, item_name: str, item_data: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
+        pass
```

Comment:
For cases were these method/properties are always called by the class, they should do `raise NotImplementedError`. 
Using `pass` will make those calls silently pass and yield another error much later that can be harder to analyse.

Reply:
Not true. If an abstract method is not defined always a `TypeError` is raised. Does not matter at all what is the content of the abstract method in the abstract class.

Reply:
You're right for the typical case of calling `STACpopulatorBase()` directly.
However, it doesn't behave the same depending on how the `ABC` gets called.

```python 
import abc

class BaseABC(abc.ABC):
    @abc.abstractmethod
    def method(self):
        pass

class BaseRaise(object):  # doesn't "require" ABC, but can be added as extra validation
    @abc.abstractmethod
    def method(self):
        raise NotImplementedError

class DeriveABC(BaseABC):
    def method(self):
        print("DeriveABC.method")

class DeriveRaise(BaseRaise):
    def method(self):
        print("DeriveABC.method")

BaseABC.method(DeriveABC())  # nothing printed, no error

BaseRaise.method(DeriveRaise())  # error below
```
```pytb
Traceback (most recent call last):
  File "/home/francis/dev/conda/envs/weaver/lib/python3.10/site-packages/IPython/core/interactiveshell.py", line 3526, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-9-f234a81a6388>", line 1, in <module>
    BaseRaise.method(DeriveRaise())
  File "<ipython-input-5-d3d4d159a3cb>", line 4, in method
    raise NotImplementedError
NotImplementedError

```

Sometimes, the `BaseABC.method(DeriveABC())` format is needed to handle specific MRO conditions (such as when I was overriding the classes in my notebook). In that case, it would not flag that the abstract class was called directly, probably not something that was done intentionnaly.

Anyway. 
I believe `raise NotImplementedError` is much more indicative of the intension "you must implement this", and if somehow `ABC` is omitted, it will still report the error.

Reply:
Okay interesting use case..... alright, will change to raising error.